apps/vs-code-designer/src/app/utils/funcCoreTools/getFuncPackageManagers.ts

```diff
@@ -2,36 +2,73 @@
+} from '../packageManagers/getBrewPackageName';
 import { executeCommand } from './cpUtils';
 import { FuncVersion } from '@microsoft/vscode-extension';
 
 /**
  * Gets package managers installed in the system.
+ * Determines which package manager installed functions core tools.
  * @param {boolean} isFuncInstalled - Is functions core tools installed.
  * @returns {Promise<PackageManager[]>} Returns array of package managers.
  */
 export async function getFuncPackageManagers(isFuncInstalled: boolean): Promise<PackageManager[]> {
   const result: PackageManager[] = [];
-  switch (process.platform) {
-    case 'linux':
-      // https://github.com/Microsoft/vscode-azurefunctions/issues/311
-      break;
-    case 'darwin':
-      if (await hasBrew(isFuncInstalled)) {
+  if (isFuncInstalled) {
+    if (process.platform == Platform.windows) {
+      const response = await executeCommand(undefined, undefined, 'winget', 'list', wingetFuncPackageName);
+      // winget response includes headers if package is installed
+      if (response.length > 2) {
+        result.push(PackageManager.winget);
+      }
+    }
+
+    if (process.platform == Platform.mac) {
+      if (
+        isBrewPackageInstalled(getFuncCoreToolsBrewPackageName(FuncVersion.v4)) ||
+        isBrewPackageInstalled(getFuncCoreToolsBrewPackageName(FuncVersion.v3)) ||
+        isBrewPackageInstalled(getFuncCoreToolsBrewPackageName(FuncVersion.v2)) ||
+        isBrewPackageInstalled(getFuncCoreToolsBrewPackageName(FuncVersion.v1))
+      ) {
         result.push(PackageManager.brew);
       }
-    // fall through to check npm on both mac and windows
-    default:
-      try {
-        isFuncInstalled
-          ? await executeCommand(undefined, undefined, 'npm', 'ls', '-g', funcPackageName)
-          : await executeCommand(undefined, undefined, 'npm', '--version');
-        result.push(PackageManager.npm);
-      } catch (error) {
-        // an error indicates no npm
+    }
+    // https://github.com/Microsoft/vscode-azurefunctions/issues/311
+    // fall through to check npm on windows, mac and linux
+    try {
+      await executeCommand(undefined, undefined, 'npm', 'ls', '-g', funcPackageName);
+      result.push(PackageManager.npm);
+    } catch (error) {
+      // an error indicates no npm
+    }
+  } else {
+    if (process.platform == Platform.windows) {
+      result.push(PackageManager.winget);
+    }
+
+    if (process.platform == Platform.mac) {
+      if (hasBrew(isFuncInstalled)) {
+        result.push(PackageManager.brew);
       }
+    }
+
+    if (process.platform == Platform.linux) {
+      // https://github.com/Microsoft/vscode-azurefunctions/issues/311
+      result.push(PackageManager.wget);
+    }
+
+    // fall through to check npm on windows, mac and linux
+    try {
+      await executeCommand(undefined, undefined, 'npm', '-v');
+      result.push(PackageManager.npm);
+    } catch (error) {
+      // an error indicates no npm
+    }
   }
   return result;
```

Comment:
nit: I would log the result (as well as where we push to that result) as well so we have telemetry if this fails for some reason.

Reply:
I think in this case - it's ok not to log as it is checking if npm is installed and adding it to the array if it does.
If there is not npm, there are no "consequences".
"Fall through" to check npm because it is common pakagemanager between Operating Systems.
The nodejs dependency is also supposed to be optional for logic app extension but it is the preferred packagemanager for azure funtions core tools